dataproc/python-api-walkthrough.md

```diff
@@ -0,0 +1,158 @@
+**For more information**: See [Cloud Dataproc&rarr;Use the Python Client Library](https://cloud.google.com/dataproc/docs/tutorials/python-library-example) for
+an explanation of how the code works.
+
+**To reload this walkthrough:** Run the following command from the
+`~/python-docs-samples/dataproc` directory in Cloud Shell:
+
+    cloudshell launch-tutorial python-api-walkthrough.md
+
+**To copy and run commands**: Click the "Paste in Cloud Shell" button
+  (<walkthrough-cloud-shell-icon></walkthrough-cloud-shell-icon>)
+  on the side of a code box, then press `Enter` to run the command.
+
+## Prerequisites (1)
+
+1. Create or select a Google Cloud Platform project to use for this tutorial.
+    * <walkthrough-project-billing-setup permissions=""></walkthrough-project-billing-setup>
+
+1. Enable the Cloud Dataproc, Compute Engine, and Cloud Storage APIs in your project.
+    * <walkthrough-enable-apis apis="dataproc,compute_component,storage-component.googleapis.com"></walkthrough-enable-apis>
+
+## Prerequisites (2)
+
+1. This walkthrough uploads a PySpark file (`pyspark_sort.py`) to a
+   [Cloud Storage bucket](https://cloud.google.com/storage/docs/key-terms#buckets) in
+   your project.
+   * You can use the [Cloud Storage browser page](https://console.cloud.google.com/storage/browser)
+   in Google Cloud Platform Console to view existing buckets in your project.
+   * To create a new bucket, run the following command. Your bucket name must be unique.
+   ```bash
+   gsutil mb -p {{project-id}} gs://your-bucket-name
+   ```
+
+1.  Set environment variables.
+
+    * Set the name of your bucket.
+    ```bash
+    BUCKET=your-bucket-name
+    ```
+
+## Prerequisites (3)
+
+1. Set up a Python
+   [virtual environment](https://virtualenv.readthedocs.org/en/latest/)
+   in Cloud Shell.
+
+    * Create the virtual environment.
+    ```bash
+    virtualenv ENV
+    ```
+    * Activate the virtual environment.
+    ```bash
+    source ENV/bin/activate
+    ```
+
+1. Install library dependencies in Cloud Shell.
+    ```bash
+    pip install -r requirements.txt
+    ```
+
+## Create a cluster and submit a job
+
+1. Set a name for your new cluster.
+    ```bash
+    CLUSTER=new-cluster-name
+    ```
+
+1. Set a [zone](https://cloud.google.com/compute/docs/regions-zones/#available)
+   where your new cluster will be located. You can change the
+   "us-central1-a" zone that is pre-set in the following command.
+    ```bash
+    ZONE=us-central1-a
+    ```
+
+1. Run `submit_job.py` with the `--create_new_cluster` flag
+   to create a new cluster and submit the `pyspark_sort.py` job
+   to the cluster.
+    ```bash
+    python submit_job_to_cluster.py \
+    --project_id={{project-id}} \
+    --cluster_name=$CLUSTER \
+    --zone=$ZONE \
+    --gcs_bucket=$BUCKET \
+    --create_new_cluster
+    ```
+
+## Job Output
+
+Job output in Cloud Shell shows cluster creation, job submission,
+    job completion, and then tear-down of the cluster.
+
+     ```bash
```

Comment:
This isn't formatting right in Cloud Shell (I see the "```bash")

Reply:
Fixed.